aptos-move/block-executor/src/executor.rs

```diff
@@ -438,46 +406,32 @@ where
         // TODO: should not need to pass base view.
         base_view: &S,
         shared_counter: &AtomicU32,
-        role: CommitRole,
+        txn_fee_state: &Mutex<(FeeStatement, Vec<FeeStatement>)>,
     ) {
         // Make executor for each task. TODO: fast concurrent executor.
         let init_timer = VM_INIT_SECONDS.start_timer();
         let executor = E::init(*executor_arguments);
         drop(init_timer);
 
-        let committing = matches!(role, CommitRole::Coordinator(_));
-
         let _timer = WORK_WITH_TASK_SECONDS.start_timer();
         let mut scheduler_task = SchedulerTask::NoTask;
-        let mut worker_idx = 0;
 
-        let mut accumulated_fee_statement = FeeStatement::zero();
-        let mut txn_fee_statements = Vec::with_capacity(block.len());
         loop {
-            // Only one thread does try_commit to avoid contention.
-            match &role {
-                CommitRole::Coordinator(post_commit_txs) => {
-                    self.coordinator_commit_hook(
-                        self.maybe_block_gas_limit,
-                        scheduler,
-                        post_commit_txs,
-                        &mut worker_idx,
-                        &mut scheduler_task,
-                        last_input_output,
-                        &mut accumulated_fee_statement,
-                        &mut txn_fee_statements,
-                    );
-                },
-                CommitRole::Worker(rx) => {
-                    while let Ok(txn_idx) = rx.try_recv() {
-                        self.worker_commit_hook(
-                            txn_idx,
-                            versioned_cache,
-                            last_input_output,
-                            base_view,
-                        );
-                    }
-                },
+            while scheduler.should_coordinate_commits() {
```

Comment:
If you consider it appropriate, maybe we can actually comment here saying that while ensures that any task that can be committed (e.g. due to a validation the worker did right before) gets committed before moving on, or something to that extent.

Reply:
Yep - a comment is appropriate. Tho, `before moving on` is vague (and possibly incorrect - some other thread might be `moving on`)